backend/src/nodes/nodes/external_stable_diffusion/img2img.py

```diff
@@ -0,0 +1,104 @@
+from __future__ import annotations
+
+import numpy as np
+
+from . import category as ExternalStableDiffusionCategory
+from ...impl.external_stable_diffusion import (
+    decode_base64_image,
+    SamplerName,
+    STABLE_DIFFUSION_IMG2IMG_URL,
+    post,
+    encode_base64_image,
+)
+from ...node_base import NodeBase, group
+from ...node_factory import NodeFactory
+from ...properties.inputs import (
+    TextInput,
+    NumberInput,
+    SliderInput,
+    EnumInput,
+    ImageInput,
+)
+from ...properties.outputs import ImageOutput
+from typing import Optional
+
+
+@NodeFactory.register("chainner:external_stable_diffusion:img2img")
+class Img2Img(NodeBase):
+    def __init__(self):
+        super().__init__()
+        self.description = "Modify an image using Automatic1111"
+        self.inputs = [
+            ImageInput(),
+            TextInput("Prompt", default="an astronaut riding a horse"),
+            TextInput("Negative Prompt").make_optional(),
+            SliderInput(
+                "Denoising Strength",
+                minimum=0,
+                default=0.75,
+                maximum=1,
+                slider_step=0.01,
+                controls_step=0.1,
+                precision=2,
+            ),
+            group("seed")(
+                NumberInput("Seed", minimum=0, default=42, maximum=4294967296)
+            ),
+            SliderInput("Steps", minimum=1, default=20, maximum=150),
+            EnumInput(SamplerName, default_value=SamplerName.EULER),
+            SliderInput(
+                "CFG Scale",
+                minimum=1,
+                default=7,
+                maximum=20,
+                controls_step=0.1,
+                precision=1,
+            ),
+            SliderInput("Width", minimum=64, default=512, maximum=2048).with_id(8),
+            SliderInput("Height", minimum=64, default=512, maximum=2048).with_id(9),
+            TextInput("Model Checkpoint Override").make_optional(),
+        ]
+        self.outputs = [
+            ImageOutput(
+                image_type="Image {width: Input8, height: Input9, channels: 3}"
```

Comment:
Using `channels: x` inside the type expression does work, but we have something better. `ImageOutput` also has a [`channels` parameter](https://github.com/chaiNNer-org/chaiNNer/blob/main/backend/src/nodes/properties/outputs/numpy_outputs.py#L42). This parameter not only adds the channel number to the type, it also enforces it at runtime. So if the API returns something that is not an RGB image, the user will see an error saying that the node is implemented incorrectly. This is important because chainner uses type information to verify the correctness of chains, so the type information has to be correct.

Same for text to image.

```suggestion
                image_type="Image { width: Input8, height: Input9 }",
                channels=3,
```